### PR TITLE
Add language support in the backend

### PR DESCRIPTION
**Pull request by neumantm**
For the strings passed back through the api and for the mail

Note this is based on #19
TODO: 
 * [x] Merge #19 and rebase this on master

### CODE REVIEW

File: ms365_accountcreator/api/account_creation.py [hunk 1]
```diff
@@ -30,13 +31,15 @@ def post(self):
...
-            LOGIC.create_account(** request.get_json())
-            return "", 204
-        except EmailIllegalError:
-            abort(400, "Email illegal")
-        except EmailAlreadyUsedError:
-            abort(400, "Email already used")
-        except GraphApiError as err:
-            abort(500, "Upstream API error")
-            APP_LOGGER.error(err)
+        lang: str = request.headers.get('lang', 'en')
```
Comment (buehlefs) [inline ms365_accountcreator/api/account_creation.py]: Maybe use `request.accept_languages.best_match(['de', 'en'])` to determine the language if the lang header is not set correctly. (See `@BABEL.localeselector` in init)

File: ms365_accountcreator/api/account_creation.py [hunk 2]
```diff
@@ -30,13 +31,15 @@ def post(self):
...
+        with force_locale(lang):
```
Comment (buehlefs) [inline ms365_accountcreator/api/account_creation.py]: Is forcing this really necessary? The same pattern (augmenting the accepted languages of the request) as in the default route in routes.py should also work.

File: ms365_accountcreator/api/account_creation.py (hunk 1 above)
Comment (neumantm) [reply ms365_accountcreator/api/account_creation.py]: This is not needed when using the same pattern as in routes.py

File: ms365_accountcreator/api/account_creation.py (hunk 2 above)
Comment (neumantm) [reply ms365_accountcreator/api/account_creation.py]: ok

**Review by buehlefs — APPROVED** [review]

**Pull request merged**